### PR TITLE
Update to Girder 2.0

### PR DESCRIPTION
**Pull request by jbeezley**
This is still kind of a WIP.  For one, I have taken out all of the annotation code.  WIth the annotation work from @manthey on GeoJS, I'm not sure it worth going through the effort to update it.

The client test is also failing on travis, and I'm at a loss for why.  It passes for me locally, but on travis, there is an [error](https://travis-ci.org/DigitalSlideArchive/large_image/builds/169923161#L5141):

```
ERROR: ReferenceError: Can't find variable: webpackJsonp
```

The build takes quite a while to complete so it is combersome to debug.  It looks like a problem with the webpack bundle, so I'm wondering if @sebastienbarre or @zachmullen have any ideas?

### CODE REVIEW

File: plugin_tests/client/largeImageSpec.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
+/* globals girder, girderTest, describe, it, expect, waitsFor, runs */
+
+girderTest.addCoveredScripts([
+    '/clients/web/static/built/plugins/jobs/plugin.min.js',
+    '/clients/web/static/built/plugins/worker/plugin.min.js',
```
Comment (sebastienbarre) [inline plugin_tests/client/largeImageSpec.js]: These two (`jobs`, `worker`) should probably be in a `girderTest.addScripts`, not `girderTest.addCoveredScripts`.

**Comment by sebastienbarre** [comment]
@jbeezley this looks to me as it *could* be an issue with the ordering of the scripts in the `testEnv.html` driver. The ordering is tricky to achieve and it seems that [recent changes](https://github.com/girder/girder/commits/master/clients/web/test/testEnv.pug) on `clients/web/test/testEnv.pug` might be worth investigating/reverting (`testing.min.js` was moved down, for example). Hope this helps.

File: plugin_tests/client/largeImageSpec.js (hunk 1 above)
Comment (jbeezley) [inline plugin_tests/client/largeImageSpec.js]: `addCoveredScripts` is just an [alias](https://github.com/girder/girder/blob/master/clients/web/test/testUtils.js#L639) of `addScripts`.

**Comment by jbeezley** [comment]
Unfortunately, the prior version of `testEnv.pug` exhibited the same behavior.

**Comment by zachmullen** [comment]
@jbeezley I'll take a look hopefully today or tomorrow. I have not seen that error message before.

Is this based on girder from before or after the phantom 2.0 upgrade?

**Comment by jbeezley** [comment]
I tried both.  I can work on it some more, I just wanted to find out if there was something obvious that I was missing.

**Comment by jbeezley** [comment]
Uploaded the built javascript files to [girder](https://data.kitware.com/#user/55087c098d777f5ff55d5e3a/folder/580e34768d777f10f26ed5e4).  Using those files, I can reproduce it locally.  I'll update when I find out more.

**Comment by jbeezley** [comment]
Debugging is less the fruitful.  One thing I did notice is that the chunking of the files is different between the two builds.  On travis, it looks like its putting the main girder code in the the large_image's file:
```
                                             Asset       Size  Chunks             Chunk Names
                       plugins/jobs/plugin.min.css    3.92 kB       2  [emitted]  plugins/jobs/plugin
              assets/jsoneditor-icons-bf23347e.png    14.4 kB          [emitted]  
 assets/glyphicons-halflings-regular-fa277232.woff    23.4 kB          [emitted]  
assets/glyphicons-halflings-regular-448c34a5.woff2      18 kB          [emitted]  
  assets/glyphicons-halflings-regular-e18bbf61.ttf    45.4 kB          [emitted]  
  assets/glyphicons-halflings-regular-89889688.svg     109 kB          [emitted]  
                                 girder.app.min.js    4.12 MB    0, 1  [emitted]  girder.app
                 plugins/large_image/plugin.min.js    1.05 MB       1  [emitted]  plugins/large_image/plugin
                        plugins/jobs/plugin.min.js     678 kB       2  [emitted]  plugins/jobs/plugin
                      plugins/worker/plugin.min.js     399 kB       3  [emitted]  plugins/worker/plugin
                                 girder.ext.min.js    2.91 MB       4  [emitted]  girder.ext
                                girder.app.min.css    45.2 kB    0, 1  [emitted]  girder.app
                                girder.ext.min.css     183 kB       4  [emitted]  girder.ext
  assets/glyphicons-halflings-regular-f4769f9b.eot    20.1 kB          [emitted]  
                plugins/large_image/plugin.min.css    5.23 kB       1  [emitted]  plugins/large_image/plugin
                     plugins/worker/plugin.min.css  257 bytes       3  [emitted]  plugins/worker/plugin
                             girder.app.min.js.map    2.49 MB    0, 1  [emitted]  girder.app
                            girder.app.min.css.map   95 bytes    0, 1  [emitted]  girder.app
             plugins/large_image/plugin.min.js.map     455 kB       1  [emitted]  plugins/large_image/plugin
            plugins/large_image/plugin.min.css.map  111 bytes       1  [emitted]  plugins/large_image/plugin
                    plugins/jobs/plugin.min.js.map     270 kB       2  [emitted]  plugins/jobs/plugin
                   plugins/jobs/plugin.min.css.map  104 bytes       2  [emitted]  plugins/jobs/plugin
                  plugins/worker/plugin.min.js.map     109 kB       3  [emitted]  plugins/worker/plugin
                 plugins/worker/plugin.min.css.map  106 bytes       3  [emitted]  plugins/worker/plugin
                             girder.ext.min.js.map    3.53 MB       4  [emitted]  girder.ext
                            girder.ext.min.css.map   95 bytes       4  [emitted]  girder.ext
```
locally, it looks like this:
```
                                             Asset       Size  Chunks             Chunk Names
                       plugins/jobs/plugin.min.css    3.92 kB       2  [emitted]  plugins/jobs/plugin
              assets/jsoneditor-icons-bf23347e.png    14.4 kB          [emitted]
 assets/glyphicons-halflings-regular-fa277232.woff    23.4 kB          [emitted]
assets/glyphicons-halflings-regular-448c34a5.woff2      18 kB          [emitted]
  assets/glyphicons-halflings-regular-e18bbf61.ttf    45.4 kB          [emitted]
  assets/glyphicons-halflings-regular-89889688.svg     109 kB          [emitted]
                                 girder.app.min.js    4.04 MB       0  [emitted]  girder.app
                 plugins/large_image/plugin.min.js    1.24 MB       1  [emitted]  plugins/large_image/plugin
                        plugins/jobs/plugin.min.js     672 kB       2  [emitted]  plugins/jobs/plugin
                      plugins/worker/plugin.min.js     397 kB       3  [emitted]  plugins/worker/plugin
                                 girder.ext.min.js    2.84 MB       4  [emitted]  girder.ext
                                girder.app.min.css    45.2 kB       0  [emitted]  girder.app
                                girder.ext.min.css     182 kB       4  [emitted]  girder.ext
  assets/glyphicons-halflings-regular-f4769f9b.eot    20.1 kB          [emitted]
                plugins/large_image/plugin.min.css    5.92 kB       1  [emitted]  plugins/large_image/plugin
                     plugins/worker/plugin.min.css  257 bytes       3  [emitted]  plugins/worker/plugin
                             girder.app.min.js.map    2.35 MB       0  [emitted]  girder.app
                            girder.app.min.css.map   95 bytes       0  [emitted]  girder.app
             plugins/large_image/plugin.min.js.map     537 kB       1  [emitted]  plugins/large_image/plugin
            plugins/large_image/plugin.min.css.map  111 bytes       1  [emitted]  plugins/large_image/plugin
                    plugins/jobs/plugin.min.js.map     260 kB       2  [emitted]  plugins/jobs/plugin
                   plugins/jobs/plugin.min.css.map  104 bytes       2  [emitted]  plugins/jobs/plugin
                  plugins/worker/plugin.min.js.map     108 kB       3  [emitted]  plugins/worker/plugin
                 plugins/worker/plugin.min.css.map  106 bytes       3  [emitted]  plugins/worker/plugin
                             girder.ext.min.js.map    3.44 MB       4  [emitted]  girder.ext
                            girder.ext.min.css.map   95 bytes       4  [emitted]  girder.ext
```

I know I've seen instances where webpack's optimization stops due to a lack of memory.  My guess is that's what accounts for the difference here, but I have no idea if that is relevant the test failure.

**Comment by sebastienbarre** [comment]
alright I'm going to put that high on my priority list. I had mentioned it in the PR (copied below), as yet to be resolved, but it seems to have adverse affect apparently:
- There still appears to be too much duplicated code going in `girder.main.min.js` and each `plugin.min.js`; this needs to be investigated and fixed by myself (Seb; consider [robertknight/webpack-bundle-size-analyzer](https://github.com/robertknight/webpack-bundle-size-analyzer)), also consider the `DllPlugin`, which would increase performance as well ([Webpack Plugins we been keepin on the DLL](https://medium.com/@soederpop/webpack-plugins-been-we-been-keepin-on-the-dll-cdfdd6cb8cd7), [Optimizing Webpack for Faster React Builds](http://engineering.invisionapp.com/post/optimizing-webpack/), [Optimizing Webpack build times and improving caching with DLL bundles](https://robertknight.github.io/posts/webpack-dll-plugins/), [webpack/examples/dll](https://github.com/webpack/webpack/tree/master/examples/dll)). Also check [trivago/parallel-webpack: Builds multi-config webpack projects in parallel](https://github.com/trivago/parallel-webpack)

**Comment by jbeezley** [comment]
It looks like the client tests now pass when running from https://github.com/girder/girder/pull/1653.  It looks like the source of the problem was that the `es2015` preset was not being used by babel for whatever reason.  The issue of duplicated code is still there, but doesn't seem to be causing any specific problems.

**Comment by manthey** [comment]
Would it be better to change this to `@import '~nib/index.styl'` like is done elsewhere in Girder 2?

**Comment by manthey** [comment]
@jbeezley When I try to view a large image in a web browser, I get a server 500 error with `TypeError: 'BaseWebroot' object is not callable`.  I don't see geojs in the static/built directory (nor is it listed as a dependency in the package.json file).

**Comment by jbeezley** [comment]
I'm not sure about the 500 error, but loading the geojs file should now be resolved.

**Comment by jbeezley** [comment]
I have no idea why `require.resolve('geojs')` is failing on travis.  It works locally with the same version of node and npm.  I'll have to do some more exploration.

**Comment by jbeezley** [comment]
I merged in the changes from master here.  The test failure is because I added codecov reporting on files that were never tested before.

**Comment by jbeezley** [comment]
Good catch.  It should be fixed now.

**Comment by manthey** [comment]
When this gets merged in, we may want to tag the most recent non Girder 2 version clearly in case we need to use it.

**Comment by Purg** [comment]
Please tag last non-2.x supported version. We are still using girder 1.7.0.

**Comment by manthey** [comment]
@purg I've tagged the most recent girder-1.7 compatible release (under tag `girder-1.7`).

**Comment by Purg** [comment]
Awesome! Thanks. :)

**Pull request merged**